flow360/component/simulation/framework/entity_base.py

```diff
@@ -0,0 +1,156 @@
+        """
+        super().__init__(**data)
+        assert self._entity_type is not None, "_entity_type is not defined in the entity class."
+        EntityRegistry.register(self)
+
+    def copy(self, update=None, **kwargs) -> EntityBase:
+        """
+        Creates a copy of the entity with compulsory updates.
+
+        Parameters:
+            update: A dictionary containing the updated attributes to apply to the copied entity.
+            **kwargs: Additional arguments to pass to the copy constructor.
+
+        Returns:
+            A copy of the entity with the specified updates.
+        """
+        if update is None:
+            raise ValueError(
+                "Change is necessary when copying an entity as there cannot be two identical entities at the same time. Please use update parameter to change the entity attributes."
+            )
+        if "name" not in update or update["name"] == self.name:
+            raise ValueError(
+                "Copying an entity requires a new name to be specified. Please provide a new name in the update dictionary."
+            )
+        super().copy(update=update, **kwargs)
+
+
+class _CombinedMeta(type(Flow360BaseModel), type):
+    pass
+
+
+class _EntitiesListMeta(_CombinedMeta):
+    def __getitem__(cls, entity_types):
+        """
+        Creates a new class with the specified entity types as a list of stored entities.
+        """
+        if not isinstance(entity_types, tuple):
+            entity_types = (entity_types,)
+        union_type = Union[entity_types]
+        annotations = {"stored_entities": List[union_type]}
+        new_cls = type(
+            f"{cls.__name__}[{','.join([t.__name__ for t in entity_types])}]",
+            (cls,),
+            {"__annotations__": annotations},
+        )
+        return new_cls
+
+
+class EntityList(Flow360BaseModel, metaclass=_EntitiesListMeta):
+    """
+    Represents a collection of volume entities in the Flow360 simulation.
+
+    Attributes:
+        stored_entities (List[Union[Box, Cylinder, str]]): List of stored entities, which can be
+            instances of `Box`, `Cylinder`, or strings representing naming patterns.
+
+    Methods:
+        format_input_to_list(cls, input: List) -> dict: Class method that formats the input to a
+            dictionary with the key 'stored_entities'.
+        check_duplicate_entity_in_list(cls, values): Class method that checks for duplicate entities
+            in the list of stored entities.
+        get_expanded_entities(self): Method that processes the stored entities to resolve any naming
+            patterns into actual entity references, expanding and filtering based on the defined
+            entity types.
+
+    """
+
+    stored_entities: List = pd.Field()
+
+    @pd.model_validator(mode="before")
+    @classmethod
+    def format_input_to_list(cls, input: List) -> dict:
+        return dict(stored_entities=input)
```

Comment:
here we don't have any validation on `input`, I think. How do we know it is a list? Also, when user provides single string (without []) we can parse it as list for user convenience

Reply:
Added validation